### ISSUE TITLE
Compiler does not warn about incorrect reference passing

### ISSUE THREAD

**Issue by leon-ts** [X-Sharp/XSharpPublic#1295]
There is the following code, where instead of OUT the reference is passed as REF:
```
FUNCTION Start() AS VOID STRICT

	LOCAL x := "" AS STRING
	TestFunc(REF x)
	
	RETURN

FUNCTION TestFunc(x OUT STRING) AS VOID STRICT
	x := "test"
	RETURN
```
The compiler generates a call using OUT (this can be seen in the IL code) without errors or warnings. But in my opinion it would be more correct in this case to issue an error or at least a warning.

**Comment by RobertvanderHulst** [comment]
Leonid,
Vulcan did not have OUT. So users that have used Vulcan before use REF instead of OUT.
I can make this a warning for the other dialects.

**Comment by cpyrgas** [comment]
Confirmed fixed

**Comment by leon-ts** [comment]
Robert, Chris,
The compiler 2.18.0.3 gives this new warning for functions, but not for class methods. In the ticket description I did not specify functions or methods, but in the example I really only provided a function. This is not a critical issue. Should I create a new ticket or will you try to fix this in 2.18?

**More complete example:**
[XSharpBetaTest.zip](https://github.com/X-Sharp/XSharpPublic/files/13187314/XSharpBetaTest.zip)

**Environment:**
Compiler X# 2.18.0.3
IDE VS 2022

**Comment by RobertvanderHulst** [comment]
Fixed in 2.18.0.4
